routers/api/v1/api.go

```diff
@@ -758,6 +758,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 				}, reqRepoReader(models.UnitTypeCode))
 				m.Group("/git", func() {
 					m.Group("/commits", func() {
+						m.Get("", repo.GetAllCommits)
```

Comment:
GitHub actually does this as `/repos/:owner/:repo/commits`, and doesn't have a listing for `GET /repos/:owner/:repo/git/commits`. Maybe we should too? (https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository)

Reply:
I think we can do a broken change on v1.10 to follow github's API.

Reply:
@Mikescher Hopefully this route path can be changed. I believe there is a placeholder for `:owner/:repo/commits` as we also need the single commit for the non-git route, as that one is to show the files that have been changed (actually have developers that would like this so their app can quickly see if anything changed since last commit and what without having a git client and without having to clone a repo), so if you can put your m.Get("", ...) call there, we can then get one that takes a SHA as well.

Reply:
Okay I changed the api path from "/repos/{owner}/{repo}/git/commits" to "/repos/{owner}/{repo}/commits".  
I think that's what you meant?

Reply:
@Mikescher yeah, perfect!!